cmd/thanos/sidecar.go

```diff
@@ -125,6 +128,31 @@ func runSidecar(
 
 		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(func() error {
+			// Only check Prometheus's flags when upload is enabled.
+			if uploadCompacted {
```

Comment:
This flag is for a different purpose :smile: this one is `true` if we want to uploads which are *already* compacted. We want to actually check this: https://github.com/improbable-eng/thanos/blob/f800a367cddd0150ba94ec0514ece9f4e59814de/cmd/thanos/sidecar.go#L265. Could you move that check over to here? :smiley:

Reply:
Thx. Can you review again?